kytos/core/tcp_server.py

```diff
@@ -44,113 +40,108 @@ def __init__(self, server_address, RequestHandlerClass,
+        This method loops reading the binary data from the connection socket,
+        and placing a 'kytos/core.messages.new' KytosEvent in the raw event
+        buffer.
         """
         curr_thread = current_thread()
-        header_len = 8
+        MAX_SIZE = 2**16
         while True:
-            header = Header()
-            binary_data, raw_header = b'', b''
             try:
-                while len(raw_header) < header_len:
-                    remaining = header_len - len(raw_header)
-                    raw_header += self.request.recv(remaining)
-                    if not raw_header:
-                        break
-            except InterruptedError as exception:
+                new_data = self.request.recv(MAX_SIZE)
+            except (SocketError, OSError, InterruptedError) as exception:
                 self.exception = exception
+                log.debug('Socket handler exception while reading: %s',
+                          exception)
+                break
+            if new_data == b'':
+                self.exception = 'Null Request (closed by client)'
                 break
-            else:
-                if not raw_header:
-                    break
 
             log.debug("New message from %s:%s at thread %s", self.ip,
                       self.port, curr_thread.name)
 
-            header.unpack(raw_header)
-            body_len = header.length - header_len
-            log.debug('Reading the binary_data')
-            try:
-                while len(binary_data) < body_len:
-                    remaining = body_len - len(binary_data)
-                    binary_data += self.request.recv(remaining)
-            except (SocketError, OSError) as exception:
-                self.exception = exception
-                break
-
             content = {'source': self.connection,
-                       'header': header,
-                       'binary_data': binary_data}
-
-            event = KytosEvent(name='kytos/core.messages.openflow.new',
+                       'new_data': new_data}
+            event = KytosEvent(name='kytos/core.server.data.in',
                                content=content)
-
             self.server.controller.buffers.raw.put(event)
 
     def finish(self):
         """Method is called when the client connection is finished.
 
-        When this method is called the request is closed, a new KytosEvent is
-        built with kytos/core.connection.lost name and put this in a app
-        buffer.
+        This method closes the connection socket and generates a
+        'kytos/core.connection.lost' KytosEvent in the App buffer.
         """
-        log.info("Client %s:%s disconnected. Reason: %s",
+        log.info("Connection lost with Client %s:%s. Reason: %s",
                  self.ip, self.port, self.exception)
+        self.connection.state = self.connection.STATES.FINISHED
+        self.request.shutdown(SHUT_WR)
```

Comment:
I'm implementing it, but first I want to be sure where it makes sense to call the shutdown and where it makes sense to call close directly.